src/org/zaproxy/zap/extension/script/PACScript.java

```diff
@@ -0,0 +1,849 @@
+                                                   "isInNet", "isPlainHostName", "isResolvable", "localHostOrDomainIs",
+                                                   "myIpAddress", "shExpMatch", "timeRange", "weekdayRange"};
+    private String scriptContent;
+    private String scriptHostName;
+
+    @Override
+    public String getClassName () {
+        return this.getClass().getName();
+    }
+
+    /**
+     * Retrieves the content pointed by {@code scriptURL} URL and creates a new PACScript object.
+     *
+     * @param scriptURL the URL corresponding to the PAC script location
+     */
+    public PACScript (URL scriptURL)
+    {
+        try {
+            scriptContent = IOUtils.toString(scriptURL.openStream());
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+        }
+        scriptHostName = scriptURL.getHost();
+    }
+
+    /**
+     * Call the FindProxyForURL function of this PAC script through the Rhino Javascript engine. <br><br>
+     *
+     * Firstly, it uses the Rhino API to bind predefined PAC functions to the execution context of this script.
+     * Secondly, it concatenates the Javascript call to FindProxyForURL to the end of the content of this script.
+     * Finally, it evaluates this script with the Rhino Engine. <br><br>
+     *
+     * The Rhino external library is used in order to make this code compliant to Java 7 and Java 8.
+     *
+     * @param destURL   the url param of FindProxyForURL(url, host)
+     * @param destHost  the host param of FindProxyForURL(url, host)
+     * @return          a String representation of the return value of FindProxyForURL(url, host)
+     *
+     * @see org.mozilla.javascript
+     */
+    public String evaluate (String destURL, String destHost)
+    {
+        Context scriptContext = Context.enter();
+        try {
+            this.defineFunctionProperties(PAC_FUNCTIONS, this.getClass(), ScriptableObject.DONTENUM);
+            Scriptable scriptScope = scriptContext.initStandardObjects(this);
+            scriptContent = scriptContent.concat(";FindProxyForURL(\"" + destURL + "\", \"" + destHost + "\")");
+            Object scriptRes = scriptContext.evaluateString(scriptScope, scriptContent, scriptHostName, 1, null);
+            return Context.toString(scriptRes);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            return "";
+        } finally {
+            Context.exit();
+        }
+    }
+
+    private static boolean isDay (String value)
+    {
+        int day;
+        try {
+            day = Integer.valueOf(value);
+        } catch (NumberFormatException e) {
+            return false;
+        }
+
+        return day > 0 && day <= 31;
+    }
+
+    private static boolean isMonth (String value)
+    {
+        String[] months = {"JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"};
```

Comment:
Thus series of strings seems to be declared more than once. Would a constant be more appropriate?